### PR TITLE
introduced excludebinary flag

### PR DESCRIPTION
**Pull request by typekpb**
fix for #369

### CODE REVIEW

File: src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java [hunk 1]
```diff
@@ -174,6 +179,87 @@ public void testStoredProcedureWithNullInputParameter() throws SQLException {
...
+      // then
+      assertTrue(getLastLogEntry().contains("{call TEST_PROC_BINARY(?,?,?)} @p2:2000, @p1:'[binary]'"));
+    } finally {
+      P6LogOptions.getActiveInstance().setExcludebinary(original);
+    }
+  }
+  
+  @Test
+  public void binaryExcludedFalse() throws SQLException {
+    boolean original = P6LogOptions.getActiveInstance().getExcludebinary();
```
Comment (felixbarny) [inline src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java]: Put into @Before method

File: src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java [hunk 2]
```diff
@@ -174,6 +179,87 @@ public void testStoredProcedureWithNullInputParameter() throws SQLException {
...
+      call.execute();
+      // out vals not logged anyway https://github.com/p6spy/p6spy/issues/133
+//      byte[] retVal = call.getBytes(resultParamName);
+//      assertEquals("foo", retVal);
+      call.close();
+  
+      // then
+      assertTrue(getLastLogEntry().contains("{call TEST_PROC_BINARY(?,?,?)} @p2:2000, @p1:'666F6F'"));
+    } finally {
+      P6LogOptions.getActiveInstance().setExcludebinary(original);
```
Comment (felixbarny) [inline src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java]: Put into @After method

File: src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java [hunk 3]
```diff
@@ -174,6 +179,87 @@ public void testStoredProcedureWithNullInputParameter() throws SQLException {
...
+    }
+  }
+
+  protected PreparedStatement getPreparedStatement(String query) throws SQLException {
+    return (connection.prepareStatement(query));
```
Comment (felixbarny) [inline src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java]: Unnecessary parenthesis

File: src/test/java/com/p6spy/engine/spy/P6TestPreparedStatement.java [hunk 4]
```diff
@@ -272,6 +275,56 @@ protected void dropPrepared(Statement statement) {
     dropPreparedStatement("drop table prepstmt_test3", statement);
   }
 
+  @Test
+  public void binaryExcludedTrue() throws SQLException {
+    boolean original = P6LogOptions.getActiveInstance().getExcludebinary();
```
Comment (felixbarny) [inline src/test/java/com/p6spy/engine/spy/P6TestPreparedStatement.java]: @before

File: src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java (hunk 1 above)
Comment (typekpb) [inline src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java]: +1, done

File: src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java (hunk 2 above)
Comment (typekpb) [inline src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java]: +1, done

File: src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java (hunk 3 above)
Comment (typekpb) [inline src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java]: +1, done

File: src/test/java/com/p6spy/engine/spy/P6TestPreparedStatement.java (hunk 4 above)
Comment (typekpb) [inline src/test/java/com/p6spy/engine/spy/P6TestPreparedStatement.java]: +1, done

**Comment by typekpb** [comment]
anyone ready to merge it, or should I do it?

**Comment by quintonm** [comment]
LGTM...

I prefer that authors merge their own branches after the code review.

**Comment by typekpb** [comment]
@quintonm OK, thanks for feedback!

**Pull request merged**